api/src/OpenTelemetry/Attributes.hs

```diff
@@ -70,7 +80,7 @@ defaultAttributeLimits =
 
 
 data Attributes = Attributes
```

Comment:
It's a shame we're already using `Attributes` as the name here, since it would arguably be a better name for what you're calling `AttributeMap`.

Reply:
Given that there are uniqueness constraints on attributes, I'm leaning towards AttributeMap somewhat so that the semantics are more obvious to consumers of the library.

> Some other implementations may use a streaming approach where every SetAttribute API call immediately results in that individual attribute value being exported using a streaming wire protocol. In such cases the enforcement of uniqueness will likely be the responsibility of the recipient of this data.

This bit of the spec is interesting though. I am trying to think of if there is any real benefit to that approach / whether that influences naming here in any way if we wanted to support that... but no issues come to mind.

Reply:
Maybe it is not necessary that `Attributes` has `attributesCount` and `attributesDropped` and another has, but I must read the spec more.